### PR TITLE
Adjusted default # of threads of graphd to # of CPU cores

### PR DESCRIPTION
**Pull request by dutor**
As title.

### CODE REVIEW

File: src/daemons/GraphDaemon.cpp [hunk 1]
```diff
@@ -110,10 +110,24 @@ int main(int argc, char *argv[]) {
         return EXIT_FAILURE;
     }
 
+    if (FLAGS_num_netio_threads == 0) {
+        FLAGS_num_netio_threads = std::thread::hardware_concurrency();
+    }
     if (FLAGS_num_netio_threads <= 0) {
         LOG(WARNING) << "Number of networking IO threads should be greater than zero";
         return EXIT_FAILURE;
     }
```
Comment (whitewum) [inline src/daemons/GraphDaemon.cpp]: combine these two conditions ?

My thought: FLAGS_num_netio_threads <= 0 means all cores

File: src/daemons/GraphDaemon.cpp [hunk 2]
```diff
@@ -110,10 +110,24 @@ int main(int argc, char *argv[]) {
...
+    LOG(INFO) << "Number of networking IO threads: " << FLAGS_num_netio_threads;
+
+    if (FLAGS_num_worker_threads == 0) {
+        FLAGS_num_worker_threads = std::thread::hardware_concurrency();
+    }
+    if (FLAGS_num_worker_threads <= 0) {
+        LOG(WARNING) << "Number of worker threads should be greater than zero";
+        return EXIT_FAILURE;
+    }
```
Comment (whitewum) [inline src/daemons/GraphDaemon.cpp]: ditto?

File: src/daemons/GraphDaemon.cpp (hunk 1 above)
Comment (dutor) [reply src/daemons/GraphDaemon.cpp]: `0` is to be a legal option and the only option to indicate _number of cores_, while a negative value is not. The second check is to cover the negative AND the failure case(`hard_concurrency` has chances to fail)

File: etc/nebula-graphd.conf.default [hunk 3]
```diff
@@ -36,10 +36,10 @@
 --session_idle_timeout_secs=60000
 # The number of threads to accept incoming connections
 --num_accept_threads=1
```
Comment (yixinglu) [inline etc/nebula-graphd.conf.default]: Miss this option?

File: src/graph/GraphFlags.cpp [hunk 4]
```diff
@@ -16,7 +16,7 @@ DEFINE_int32(session_reclaim_interval_secs, 10, "Period we try to reclaim expire
 DEFINE_int32(num_netio_threads, 0,
                 "Number of networking threads, 0 for number of physical CPU cores");
 DEFINE_int32(num_accept_threads, 1, "Number of threads to accept incoming connections");
```
Comment (yixinglu) [inline src/graph/GraphFlags.cpp]: ditto

File: etc/nebula-graphd.conf.default (hunk 3 above)
Comment (dutor) [reply etc/nebula-graphd.conf.default]: > Miss this option?

Nope. One single thread is enough to handle incomming connections.

**Pull request merged**